addons/pos_self_order/models/pos_session.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api, _, fields
+
+
+class PosSession(models.Model):
+    _inherit = 'pos.session'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        sessions = super(PosSession, self).create(vals_list)
+        date_string = fields.Date.today().isoformat()
+        ir_sequence = self.env['ir.sequence'].sudo().search([('code', '=', f'pos.order_{date_string}')])
+
+        for session in sessions:
+            session.env['ir.sequence'].sudo().create({
+                'name': _("PoS Order by Session"),
+                'padding': 4,
+                'code': f'pos.order_{session.id}',
+                'number_next': 1,
+                'number_increment': 1,
+                'company_id': session.config_id.company_id.id,
+            })
+
+        if not ir_sequence:
+            self.env['ir.sequence'].sudo().create({
+                'name': _("PoS Order"),
+                'padding': 0,
+                'code': f'pos.order_{date_string}',
+                'number_next': 1,
+                'number_increment': 1,
+                'company_id': session.config_id.company_id.id,
+            })
```

Comment:
bad indentation there

Reply:
There is this code at the beginning of the function
```
ir_sequence = self.env['ir.sequence'].sudo().search([('code', '=', f'pos.order_{date_string}')])
```

We need to create one of these ir.sequence for all sessions